File: config/libs/react.lua
local ____lualib = require("lualib_bundle")
local __TS__ArrayFilter = ____lualib.__TS__ArrayFilter
local __TS__ArrayFlat = ____lualib.__TS__ArrayFlat
local ____exports = {}
local ____View = require("config.libs.nativeComponents.View")
local View = ____View.View
function ____exports.createElement(component, props, ...)
    local children = {...}
    local element = component(
        {children = children},
        table.unpack({props})
    )
    return __TS__ArrayFlat(
        {
            element,
            component == View and ({}) or __TS__ArrayFilter(
                children,
                function(____, child) return child end
            )
        },
        math.huge
    )
end
function ____exports.Fragment()
    return {}
end
return ____exports

File: src/config/libs/react.ts
import { View } from "./nativeComponents/View";

export function createElement(component: Function, props?: Object, ...children: any[]) {
    // const element = component(props);
    const element = component.apply({ children }, [props]);
    // console.log({ component, props, children, element });
    console.log({ component, props, children: JSON.stringify(children), element: JSON.stringify(element) });
    // filter child for && condition component
    // return [element, children.filter((child) => child)].flat(Infinity);
    return [element, component === View ? [] : children.filter((child) => child)].flat(Infinity);
    // return element;
    // return [element];
}

export function Fragment() {
    return [];
    // return [{component: 'Fragment'}];
}
